generated/nidmm/nidmm_service.cpp

```diff
@@ -406,8 +406,9 @@ namespace nidmm_grpc {
           break;
       }
 
-      auto status = library_->CloseExtCal(vi, action);
-      response->set_status(status);
+      auto cleanup_lambda = [&](uint32_t id) { library_->CloseExtCal(id, action); };
```

Comment:
We should not register new cleanup lambdas for this implementation. We would be adding the mapping from `vi` to the lambda only to remove it immediately afterwards. I think it makes more sense to call `session_repository_->access_session`, and return an error if the `vi` wasn't found. Then we need to remove the

Reply:
@jonlugoatx 
The clean up lambda function registered during InitExtCal will take 0 for action parameter. But for CloseExtCal function, clients can pass in any value for that action parameter. But if we just call remove session without updating clean up lambda, it will call the CloseExtCal with action as 0 always and action value passed by client gets ignored.

Reply:
Thanks for pointing out this use case. I hadn't considered it. Replacing the cleanup lambda is still dangerous in my opinion. I'd be more comfortable with one of these options:
1. Split `remove_session` into `close_session` which calls close lambda and `remove_session` which doesn't call the lambda.
2. Add a boolean `call_close` on `remove_session` that defaults to `true` in the header.

Then any manual call to `Close` or `CloseExtCal` would simply remove the session (no cleanup) and call the close directly afterwards. If the user resets the server, it'll do the auto cleanup.

Reply:
@jonlugoatx , I took the path of the first option you mentioned. So I removed calling of cleanup session from inside remove_session. But you had asked to create one more helper for close session right? But that helper already exists and its named as `cleanup_session` and is already being used by reset_server. So I did not add that second helper again. Can you pl review the updated changes?